### PR TITLE
Fix Array of union, TypePredicateNode, @nativeNew, = noImpl, and @nativeGetter

### PR DESCRIPTION
**Pull request by epabst**
This is based on feedback on https://github.com/bashor/ts2kt/pull/10

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -93,4 +93,12 @@
         </exec>
     </target>
 
+    <!-- This is known to fail but gives an idea of what the hot issues are. -->
+    <target name="run.test.for.testDefinitelyTyped">
```
Comment (bashor) [inline build.xml]: [non-issue]
Why You don't use idea run configurations instead?

Comment (epabst) [inline build.xml]: How?  I can add the instructions to the README.

File: testData/interface/variables/variables.d.ts [hunk 2]
```diff
@@ -5,3 +5,8 @@ interface Foo {
     varAsBoolean: boolean;
     varAsString: string;
 }
+declare namespace foo {
+    interface Bar {
+        name?: string;
```
Comment (epabst) [inline testData/interface/variables/variables.d.ts]: Fixed.

**Comment by epabst** [comment]
Updated.

File: testData/extendExternalDeclarations/withGenericParams.d.kt [hunk 3]
```diff
@@ -17,7 +17,7 @@ object ref-array {
         @nativeSetter
         fun set(prop: String, value: Number)
         var someField: String
-        var optionalField: T? = noImpl
+        var optionalField: T?
```
Comment (epabst) [inline testData/extendExternalDeclarations/withGenericParams.d.kt]: Fixed.

File: build.xml (hunk 1 above)
Comment (bashor) [inline build.xml]: 1. In the main menu: Run > Run... (Debug...)
you will get the list of available run configurations
2. Choice what you need and run

You can find more information [here](https://www.jetbrains.com/help/idea/2016.1/run-debug-configuration.html)

File: testDefinitelyTyped/expected/jquery/jquery.d.kt [hunk 4]
```diff
@@ -39,7 +39,6 @@ interface JQueryAjaxSettings {
 }
 @native
 interface JQueryXHR : XMLHttpRequest, JQueryPromise<Any> {
-    // TODO: investigate
```
Comment (bashor) [inline testDefinitelyTyped/expected/jquery/jquery.d.kt]: I'll fix it before push

File: testData/extendExternalDeclarations/withGenericParams.d.kt (hunk 3 above)
Comment (bashor) [inline testData/extendExternalDeclarations/withGenericParams.d.kt]: Nice! Thanks!

File: testData/interface/variables/variables.d.ts (hunk 2 above)
Comment (bashor) [inline testData/interface/variables/variables.d.ts]: As I see it's not fixed, but I'll fix it before push

**Comment by bashor** [comment]
Merged manually.

Thank You for your contribution!

File: build.xml (hunk 1 above)
Comment (epabst) [inline build.xml]: I'm familiar with that dialog, but I don't know kind of execution it should be with what options.  I don't know what Plugins I need to install.

**Pull request closed without merging**